plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -59,28 +61,50 @@
         }
       });
 
+  private static AnalyticsPingManager instance;
+
+  // Preference store (should be ConfigurationScope-d) from which we get UUID, opt-in status, etc.
+  private IEclipsePreferences preferences;
+
+  private OptInDialogCreator optInDialogCreator;
+
+  @VisibleForTesting
+  AnalyticsPingManager(
+      IEclipsePreferences preferences, OptInDialogCreator optInDialogCreator) {
+    this.preferences = preferences;
+    this.optInDialogCreator = optInDialogCreator;
+  }
+
+  public static AnalyticsPingManager getInstance() {
```

Comment:
also see here:

http://stackoverflow.com/questions/11165852/java-singleton-and-synchronization

singletons are much trickier than they look in multihtreaded environments